### PR TITLE
Laboratory work #4, Alina Aisina - 21FPL1

### PR DESCRIPTION
**Pull request by itsaisina**

### CODE REVIEW

File: lab_4_summarization_textrank/main.py [hunk 1]
```diff
@@ -120,23 +157,42 @@ def __init__(self) -> None:
         """
         Constructs all the necessary attributes
         """
-        pass
+        super().__init__()
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:160]: Please, check https://github.com/fipl-hse/2022-2-level-labs/blob/9045abee06da422d19ff5ed4bcb9fbb5cf587b30/lab_4_summarization_textrank/README.md?plain=1#L216

File: lab_4_summarization_textrank/start.py [hunk 2]
```diff
@@ -25,8 +26,11 @@
...
-    paths_to_texts = [str(path) for path in TEXTS_PATH.glob('*.txt')]
+    sentence_preprocessor = SentencePreprocessor(stop_words, tuple('.,!?-:;()&'))
+    sentences = sentence_preprocessor.get_sentences(text)
+    sentence_encoder = SentenceEncoder()
+    sentence_encoder.encode_sentences(sentences)
 
     RESULT = None
     # DO NOT REMOVE NEXT LINE - KEEP IT INTENTIONALLY LAST
-    assert RESULT, 'Summaries are not extracted'
+    # assert RESULT, 'Summaries are not extracted'
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/start.py:36]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/start.py (hunk 2 above)
Comment (itsaisina) [reply lab_4_summarization_textrank/start.py]: done

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -2,8 +2,12 @@
 Lab 3
 Extract keywords based on TextRank algorithm
 """
+import csv
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:5]: revert

File: lab_3_keywords_textrank/main.py [hunk 4]
```diff
@@ -41,6 +45,7 @@ class TextPreprocessor:
     preprocess_text(text: str) -> tuple[str, ...]:
         Produces filtered clean lowercase tokens from raw text
     """
+
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:48]: revert

File: lab_3_keywords_textrank/start.py [hunk 5]
```diff
@@ -1,7 +1,6 @@
 """
 TextRank keyword extraction starter
 """
-
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/start.py:4]: revert

File: lab_3_keywords_textrank/target_score.txt [hunk 6]
```diff
@@ -1 +1 @@
-8
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/target_score.txt:1]: revert

File: lab_4_summarization_textrank/main.py (hunk 1 above)
Comment (artyomtugaryov) [reply lab_4_summarization_textrank/main.py:164]: perfect

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 3 above)
Comment (itsaisina) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 4 above)
Comment (itsaisina) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/start.py (hunk 5 above)
Comment (itsaisina) [reply lab_3_keywords_textrank/start.py]: done

File: lab_3_keywords_textrank/target_score.txt (hunk 6 above)
Comment (itsaisina) [reply lab_3_keywords_textrank/target_score.txt]: done

**Review by artyomtugaryov — APPROVED** [review]

**Review by artyomtugaryov — APPROVED** [review]

File: lab_4_summarization_textrank/main.py [hunk 7]
```diff
@@ -146,7 +208,17 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
     :param other_sequence: a sequence of items
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:208]: reply contains one sentence per text

File: lab_4_summarization_textrank/main.py [hunk 8]
```diff
@@ -146,7 +208,17 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
...
     """
-    pass
+    if not isinstance(sequence, list | tuple) or not isinstance(other_sequence, list | tuple):
+        raise ValueError
+    try:
+        similar = 0
+        for elem in sequence:
+            if elem in other_sequence:
+                similar += 1
+        jaccard_index = similar / (len(set(sequence) | set(other_sequence)))
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:218]: count similar with sets as well

File: lab_4_summarization_textrank/main.py [hunk 9]
```diff
@@ -160,31 +232,51 @@ def __init__(self) -> None:
...
-        """
-        pass
+        # """
+        if not isinstance(vertex1, Sentence) or not isinstance(vertex2, Sentence) \
+                or vertex1.get_encoded() == vertex2.get_encoded():
+            raise ValueError
+        for vertex in vertex1, vertex2:
+            if vertex not in self._vertices:
+                self._vertices.append(vertex)
+                self._matrix.append([calculate_similarity(vertex.get_encoded(), other.get_encoded())
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:273]: double computations?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request merged**